azure/services/bastionhosts/bastionhosts_test.go

```diff
@@ -359,7 +306,10 @@ func TestDeleteBastionHost(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			t.Parallel()
+			// TODO(karuppiah789): Make the tests parallel and still work in case of errors and success with no false results.
```

Comment:
is this a todo for this PR?

Reply:
No, for the future I think, since I haven't found out what's causing the false report when running the tests in parallel. I have to dig into it

Reply:
I could try to dig into it as part of this PR itself. I'll check it out sometime

Reply:
Ok if it's for the future let's open a github issue instead

Reply:
I created #1760 with some details for now

Reply:
I'll resolve this conversation?

Reply:
And should I remove the `TODO` from the code?

Reply:
@karuppiah7890 did you try adding `tc := tc` right before the `t.Run` line? I believe that's what's needed for parallel tests

eg. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/groups/groups_test.go#L236

Reply:
But looking at the groups_test.go, looks like the `tc := tc` is valid and I'm confused as to how 😅

Reply:
see https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks

```
func TestGroupedParallel(t *testing.T) {
    for _, tc := range tests {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) {
            t.Parallel()
            ...
        })
    }
}
```